src/Storages/Statistics/StatisticsCountMinSketch.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <Storages/Statistics/Statistics.h>
+
+#include "config.h"
+
+#if USE_DATASKETCHES
+
+#include <count_min.hpp>
+
+namespace DB
+{
+
+class StatisticsCountMinSketch : public IStatistics
+{
+public:
+    StatisticsCountMinSketch(const SingleStatisticsDescription & stat_, DataTypePtr data_type_);
+
+    Float64 estimateEqual(const Field & val) const override;
+
+    void update(const ColumnPtr & column) override;
+
+    void serialize(WriteBuffer & buf) override;
+    void deserialize(ReadBuffer & buf) override;
+
+private:
+    using Sketch = datasketches::count_min_sketch<UInt64>;
+    Sketch sketch;
+
+    DataTypePtr data_type;
+};
+
+}
+
+#endif
+
+namespace DB
```

Comment:
I think we can put the entire header and the entire source file under ifdef `USE_DATASKETCHES`. As far as I see, all using code is in Storeages/Statistics/Statistics.cpp and properly guarded by ifdefs. Special handling using "FEATURE_IS_NOT_ENABLED_AT_BUILD_TIME" will not be needed.